cookbooks/bcpc-hadoop/recipes/journalnode.rb

```diff
@@ -32,6 +32,24 @@
     action :create
     recursive true
   end
+
+ if get_config("jn_txn_fmt") then
+   file "#{Chef::Config[:file_cache_path]}/jn_fmt.tgz" do
+     user "hdfs"
+     group "hdfs"
+     user 0644
+     content Base64.decode64(get_config("jn_txn_fmt"))
+     not_if { node[:bcpc][:hadoop][:mounts].all? { |d| File.exists?("/disk/#{d}/dfs/jn/#{node.chef_environment}/current/VERSION") } }
+   end
+ end
+
+ bash "unpack-jn-fmt-image-to-disk-#{d}" do
+   user "root"
+   cwd "/disk/#{d}/dfs/"
+   code "tar xpzvf #{Chef::Config[:file_cache_path]}/jn_fmt.tgz"
+   notifies :restart, "service[hadoop-hdfs-journalnode]"
+   only_if { not get_config("jn_txn_fmt").nil? and not File.exists?("/disk/#{d}/dfs/jn/#{node.chef_environment}/current/VERSION") }
```

Comment:
I don't see the harm in running this for an old jn_txn_fmt? I think `not get_config("jn_txn_fmt").nil? and` is likely unnecessary; if `#{Chef::Config[:file_cache_path]}/jn_fmt.tgz` exists, I say we go with it. (Throw an existence check in for the tarball?)